sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -234,7 +234,14 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       if (l.isDefined && r.isDefined) {
         b match {
           case _: Predicate =>
-            Some(new V2Predicate(b.sqlOperator, Array[V2Expression](l.get, r.get)))
+            if (isBinaryComparisonOperator(b.sqlOperator) && l.get.isInstanceOf[LiteralValue[_]] &&
+              r.get.isInstanceOf[FieldReference]) {
+              Some(new V2Predicate(flipComparisonOperatorName(b.sqlOperator),
+                Array[V2Expression](r.get, l.get)))
+            } else {
+              Some(new V2Predicate(b.sqlOperator, Array[V2Expression](l.get, r.get)))
+            }
+
```

Comment:
Let's simplify by using 4 line addition instead of 8 line code change.
```scala
        b match {
+         case _: Predicate if isBinaryComparisonOperator(b.sqlOperator) &&
+             l.get.isInstanceOf[LiteralValue[_]] && r.get.isInstanceOf[FieldReference] =>
+           Some(new V2Predicate(flipComparisonOperatorName(b.sqlOperator),
+              Array[V2Expression](r.get, l.get)))
          case _: Predicate =>
            Some(new V2Predicate(b.sqlOperator, Array[V2Expression](l.get, r.get)))
          case _ =>
            Some(new GeneralScalarExpression(b.sqlOperator, Array[V2Expression](l.get, r.get)))
        }
```

Reply:
Fixed. Thanks